File: backend/app/schemas/file.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class FileResponse(BaseModel):
    id: uuid.UUID
    conversation_id: uuid.UUID
    message_id: uuid.UUID | None = None
    original_filename: str
    file_type: str
    file_size: int
    token_estimate: int | None = None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/file.ts
export type FileResponse = {
  id: string
  conversation_id: string
  message_id: string | null
  original_filename: string
  file_type: 'pdf' | 'markdown'
  file_size: number
  token_estimate: number | null
  created_at: string
}

// Local state while the file is being uploaded
export type PendingFile = {
  localId: string          // temporary client-side id
  file: File               // browser File object
  status: 'uploading' | 'done' | 'error'
  fileResponse?: FileResponse
  error?: string
}
